types/react/index.d.ts

```diff
@@ -1098,7 +1098,7 @@ declare namespace React {
      * @version 16.8.0
      * @see https://react.dev/reference/react/useEffect
      */
-    function useEffect(effect: EffectCallback, deps?: DependencyList): void;
+    function useEffect(effect: EffectCallback, deps: DependencyList | void | null): void;
```

Comment:
This may result in the same type-checking behavior but looks confusing upon inspection. I'm also not sure TypeScript error message now correctly state that `useEffect` takes 1-2 arguments. So I'd rather stick with the standard `deps?: T` instead of `deps: T | void` unless we explicitly want to force certain type-checking behavior.

Reply:
It should not be type error to leave second argument empty.
Example: https://tsplay.dev/NrjzoN

<img width="498" alt="image" src="https://github.com/DefinitelyTyped/DefinitelyTyped/assets/12800230/6e4215db-eb86-49fc-858d-3ce471e2f553">

I guess it's better to ask React maintainers why they still have such types in repository rather than trying to fix it with typescript.

Reply:
`dependencies?` is also what we document so let's just port this 1:1 in the types